commons/src/main/java/io/hyscale/commons/framework/event/model/ActivityState.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.commons.framework.event.model;
+
+public enum ActivityState {
+    STARTED, DONE("DONE"), SKIPPING("SKIPPED"), FAILED("FAILED"), IN_PROGRESS;
```

Comment:
Is there any specific reason for keeping some are with string parameters and some are just normal enums?

Reply:
Activity message won't be necessary for every state. As you can see start and in progress won't need any message.